test/filestorage.js

```diff
@@ -0,0 +1,184 @@
+    test.error(err);
+    fs.mkdir(path.join(dir, '0003'), err => {
+      test.error(err);
+      const file = path.join(dir, '0003', '0000.f');
+      fs.writeFile(file, 'data3', err => {
+        test.error(err);
+        storage.rm(new common.Uint64(3), err => {
+          test.error(err);
+          test.strictSame(fs.existsSync(file), false);
+          test.end();
+        });
+      });
+    });
+  });
+});
+
+fsTest.test('Compress small file using zip', test => {
+  const dir = path.join(testDir, '4');
+  create({ dir, minCompressSize }, (err, storage) => {
+    test.error(err);
+    fs.mkdir(path.join(dir, '0004'), err => {
+      test.error(err);
+      fs.writeFile(path.join(dir, '0004', '0000.f'), 'data4', err => {
+        test.error(err);
+        storage.compress(new common.Uint64(4), 'ZIP', (err, compressed) => {
+          test.error(err);
+          test.strictSame(compressed, false);
+          test.end();
+        });
+      });
+    });
+  });
+});
+
+fsTest.test('Compress small file using gzip', test => {
+  const dir = path.join(testDir, '5');
+  create({ dir, minCompressSize }, (err, storage) => {
+    test.error(err);
+    test.error(err);
+    fs.mkdir(path.join(dir, '0005'), err => {
+      test.error(err);
+      fs.writeFile(path.join(dir, '0005', '0000.f'), 'data5', err => {
+        test.error(err);
+        storage.compress(new common.Uint64(5), 'GZIP', (err, compressed) => {
+          test.error(err);
+          test.strictSame(compressed, false);
+          test.end();
+        });
+      });
+    });
+  });
+});
+
+fsTest.test('Compress big file using zip', test => {
+  const data = 'data6'.repeat(1000);
+  const dir = path.join(testDir, '6');
+  const file = path.join(dir, '0006', '0000.f');
+  create({ dir, minCompressSize }, (err, storage) => {
+    test.error(err);
+    fs.mkdir(path.join(dir, '0006'), err => {
+      test.error(err);
+      fs.writeFile(file, data, err => {
+        test.error(err);
+        storage.compress(new common.Uint64(6), 'ZIP', (err, compressed) => {
+          test.error(err);
+          test.strictSame(compressed, true);
+          test.strictSame(fs.statSync(file).size < 5000, true);
+          test.end();
+        });
+      });
+    });
+  });
+});
+
+fsTest.test('Compress big file using gzip', test => {
+  const data = 'data7'.repeat(1000);
+  const dir = path.join(testDir, '7');
+  const file = path.join(dir, '0007', '0000.f');
+  create({ dir, minCompressSize }, (err, storage) => {
+    test.error(err);
+    fs.mkdir(path.join(dir, '0007'), err => {
+      test.error(err);
+      fs.writeFile(file, data, err => {
+        test.error(err);
+        storage.compress(new common.Uint64(7), 'GZIP', (err, compressed) => {
+          test.error(err);
+          test.strictSame(compressed, true);
+          test.strictSame(fs.statSync(file).size < 5000, true);
+          test.end();
+        });
+      });
+    });
+  });
+});
+
```

Comment:
Could you also add test that will `write -> compress -> read file and compare contents`?

Reply:
This is marked resolved but I don't see a corresponding test, could you point me to the line?

Just to be clear, I wanted a test that will call `storage.write(id1, buf1)` then `storage.compress(id1)` then `buf2 = storage.read(id1)` and then compare `buf1 === buf2`.

Reply:
Could you also add a simpler version of this test that will skip the compression step?